src/services/collections.js

```diff
@@ -1,19 +1,54 @@
-import config from 'config';
+import { URL } from 'url';
 
 import fetch from '../utils/fetch.js';
+import logger from '../utils/logger.js';
 
-export const fetchCollections = async () => {
-  const collections = await fetch(config.get('collectionsUrl'));
+export const fetchCollections = async collectionsConfig => {
+  const unqiueCollections = new Map();
 
-  return Object.keys(collections).reduce((result, collectionName) => {
-    if (collections[collectionName].endpoint) {
-      result.push({
-        name: collectionName,
-        id: collections[collectionName].id,
-        endpoint: collections[collectionName].endpoint,
-      });
+  const result = await Promise.allSettled(collectionsConfig.map(async item => {
+    let collections = [];
+
+    if (typeof item === 'string') {
+      collections = await fetch(item);
+    }
+
+    if (typeof item === 'object') {
+      collections = [item];
     }
 
-    return result;
-  }, []);
+    return filterInvalidCollections(collections);
+  }));
+
+  result.forEach(({ value }) => value.forEach(collection => unqiueCollections.set(collection.id, collection)));
+
+  return Array.from(unqiueCollections.values());
 };
+
+export function filterInvalidCollections(collections) {
+  return collections.filter(collection => {
+    const haveMandatoryFields = collection.id && collection.name && collection.endpoint;
```

Comment:
This is for only one collection, so singular.
```suggestion
    const hasMandatoryFields = collection.id && collection.name && collection.endpoint;
```